nbgitpuller/hookspecs.py

```diff
@@ -0,0 +1,22 @@
+import pluggy
+
+hookspec = pluggy.HookspecMarker("nbgitpuller")
+hookimpl = pluggy.HookimplMarker("nbgitpuller")
+
+
+@hookspec(firstresult=True)
+def handle_files(query_line_args):
+    """
+    :param json query_line_args: this includes any argument you put on the url
+    :return two parameter json unzip_dir and origin_repo_path
+    :rtype json object
+
+    The developer uses this function to download, un-compress and save the
+    source files to the TEMP_DOWNLOAD_REPO_DIR folder.
+
+    The parameter, query_line_args, is any argument you put on the URL
+
+    Once the files are saved to the directory, git puller can handle all the
+    standard functions needed to make sure source files are updated or created
+    as needed.
+    """
```

Comment:
It seems a bit weird to me to have handle_files have a "query_line_args" parameter declared, when we in practice call handle_files with an object that is a mash of query line args and other things in our handlers.py logic.

That makes me think the name query_line_args is a bit misleading. I wonder about the handle_files function signature a bit, should it really just take a single dict with misc key/value's, if so, perhaps we want to make it accept **kwargs which would fit the bill for a misc scrap of args.

Hmmm... I hope to learn more and I don't have a clear suggestion based on my current understanding, but the function signature of handle_files seems a bit off.

Reply:
__docstring improvements__

Initial paragraph: can you make the initial paragraph describe that this is a function signature that content provider plugins can implement, and when they do, the implementation is responsible for doing _something something_ given the arguments _something something_.

> The developer uses this function [...]

"The developer" is a bit vague to me. Can you make it clear that a _content provider plugin developer_ should define a function to do certain things in the actual implementation of this function.

Reply:
If I understand correctly as far as a user is concerned they give it a URL, and they end up with a directory of files? Is `unzip_dir` is the directory containing the output the user wants? If so you could just call it `output_dir`?

Reply:
got @manics good deal.